indexer/s3_bucket.tf

```diff
@@ -12,7 +12,8 @@ resource "aws_s3_bucket" "load_balancer" {
 # TODO: refactor snapshotting full node into a separate module
 # AWS S3 bucket to store all Indexer full node snapshots
 resource "aws_s3_bucket" "indexer_full_node_snapshots" {
-  bucket = var.s3_snapshot_bucket
+  # Use account id for mainnet to avoid name collisions
+  bucket = var.environment == "mainnet" ? "${local.account_id}-${var.s3_snapshot_bucket}" : var.s3_snapshot_bucket
```

Comment:
don't we want to use account_id regardless to avoid name collisions?

Reply:
Yes, but this would lead to plans / applies that delete existing buckets so a migration is necessary.

Reply:
nit: add a todo and create a low priority ticket on the backlog?